### DISCUSSION TITLE
Hooking into the stub publish command

### DISCUSSION THREAD

**Discussion by ollieread** [laravel/framework#33977] [General]
The stub publish command currently sits outside of `vendor:publish` so third party packages that provide customisable stubs can't hook into it.

Originally I had considered adding an event to the `stub:publish` command that collects an array of stubs, but would it not be better to have this has a group on the vendor publish command? 

I don't see a reason why the default Laravel behaviour couldn't make use of the `publishes()` method in the service providers. This would make it easier for third-party packages to use, as well as letting each individual Laravel components provide its own stubs, rather than having them in one big array.

**Comment by benjam-es** [comment]
Solution in a service provider:

```php 
use Illuminate\Console\Events\CommandFinished;
use Illuminate\Contracts\Console\Kernel;

..

$this->app->make('events')->listen(CommandFinished::class, function ($event) {
    if ($event->command == 'stub:publish') {
        $this->app->make(Kernel::class)->queue('mycommand:here');
    }
});
````
